File: Components/Reporting/CS/DevExpress.DemoReports/Data/Countries.cs
using System.Data.Common;
using System.Data.Entity;
using System.Data.Entity.Infrastructure;
using System.Data.SQLite;
using System.IO;
using DevExpress.Internal;

namespace DevExpress.DemoData.Models {
    public class CountriesEntities : DbContext {
        public CountriesEntities() : base(CreateConnection(GenerateConnectionString()), contextOwnsConnection: true) { }

        public CountriesEntities(string connectionString) : base(CreateConnection(connectionString), contextOwnsConnection: true) { }
        public CountriesEntities(DbConnection connection) : base(connection, true) { }

        static CountriesEntities() {
            Database.SetInitializer<CountriesEntities>(null);
        }

        protected override void OnModelCreating(DbModelBuilder modelBuilder) {
             modelBuilder.Entity<AboutRegion>();
             modelBuilder.Entity<Hierarchy>();
        }

        static DbConnection CreateConnection(string connectionString) {
            var connection = DbProviderFactories.GetFactory("System.Data.SQLite.EF6").CreateConnection();
            connection.ConnectionString = connectionString;
            return connection;
        }

        static string filePath;
        static string GenerateConnectionString() {
            if(filePath == null)
                filePath = DataDirectoryHelper.GetFile("countries.db", DataDirectoryHelper.DataFolderName);
            try {
                var attributes = File.GetAttributes(filePath);
                if(attributes.HasFlag(FileAttributes.ReadOnly)) {
                    File.SetAttributes(filePath, attributes & ~FileAttributes.ReadOnly);
                }
            } catch { }
            return new SQLiteConnectionStringBuilder { DataSource = filePath }.ConnectionString;
        }

        public virtual DbQuery<AboutRegion> AboutRegions { get { return Set<AboutRegion>().AsNoTracking(); } }
        public virtual DbQuery<Hierarchy> Hierarchy { get { return Set<Hierarchy>().AsNoTracking(); } }
    }
}

File: Components/Reporting/VB/DevExpress.DemoReports/Data/Countries.vb
Imports System.Data.Common
Imports System.Data.Entity
Imports System.Data.Entity.Infrastructure
Imports System.Data.SQLite
Imports System.IO
Imports DevExpress.Internal

Namespace DevExpress.DemoData.Models
    Public Class CountriesEntities
        Inherits DbContext

        Public Sub New()
            MyBase.New(CreateConnection(GenerateConnectionString()), contextOwnsConnection:= True)
        End Sub

        Public Sub New(ByVal connectionString As String)
            MyBase.New(CreateConnection(connectionString), contextOwnsConnection:= True)
        End Sub
        Public Sub New(ByVal connection As DbConnection)
            MyBase.New(connection, True)
        End Sub

        Shared Sub New()
            Database.SetInitializer(Of CountriesEntities)(Nothing)
        End Sub

        Protected Overrides Sub OnModelCreating(ByVal modelBuilder As DbModelBuilder)
             modelBuilder.Entity(Of AboutRegion)()
             modelBuilder.Entity(Of Hierarchy)()
        End Sub

        Private Shared Function CreateConnection(ByVal connectionString As String) As DbConnection
            Dim connection = DbProviderFactories.GetFactory("System.Data.SQLite.EF6").CreateConnection()
            connection.ConnectionString = connectionString
            Return connection
        End Function

        Private Shared filePath As String
        Private Shared Function GenerateConnectionString() As String
            If filePath Is Nothing Then
                filePath = DataDirectoryHelper.GetFile("countries.db", DataDirectoryHelper.DataFolderName)
            End If
            Try
                Dim attributes = File.GetAttributes(filePath)
                If attributes.HasFlag(FileAttributes.ReadOnly) Then
                    File.SetAttributes(filePath, attributes And (Not FileAttributes.ReadOnly))
                End If
            Catch
            End Try
            Return New SQLiteConnectionStringBuilder With {.DataSource = filePath}.ConnectionString
        End Function

        Public Overridable ReadOnly Property AboutRegions() As DbQuery(Of AboutRegion)
            Get
                Return [Set](Of AboutRegion)().AsNoTracking()
            End Get
        End Property
        Public Overridable ReadOnly Property Hierarchy() As DbQuery(Of Hierarchy)
            Get
                Return [Set](Of Hierarchy)().AsNoTracking()
            End Get
        End Property
    End Class
End Namespace
